packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -138,14 +141,23 @@ class InputDateTimePicker extends React.Component {
 					this.setPickerVisibility(true);
 				}
 				break;
+			case keycode.codes.enter:
+				// commiting existing change on enter
+				this.onFinish(event, 'INPUT');
```

Comment:
This is problematic because it adds additional behavior on buttons too. Each date/month/year is a button. And enter intends to press the button. Not submit the datetime-form

Furthermore, in practice, the enter close the picker and the selection is after, so it's equivalent to a cancel.